code/unit_tests/recipe_tests.dm

```diff
@@ -0,0 +1,77 @@
+/*
+ *  Unit Tests for various recipes.
+ *
+ */
+
+#define SUCCESS 1
+#define FAILURE 0
+
+datum/unit_test/research_design_cost
+	name = "RECIPES: Design Cost"
+
+datum/unit_test/research_design_cost/start_test()
+	var/tested_count = 0
+	var/error_count = 0
+	for(var/T in subtypesof(/datum/design))
+		var/datum/design/D = new T
+		if(ispath(D.build_path, /obj/item))
+			var/obj/item/I = D.Fabricate()
+			if(I.matter && D.materials && I.recyclable) // non-recyclable items can't be exploited
+				for(var/mat in I.matter)
+					tested_count++
+					if(mat in D.materials)
+						if(I.matter[mat] > D.materials[mat])
+							fail("Design '[D.name]' costs less material '[mat]' ([D.materials[mat]]) than the product is worth ([I.matter[mat]]).")
+							error_count++
+					else
+						fail("Design '[D.name]' does not require material '[mat]' even though the product is worth [I.matter[mat]].")
+						error_count++
+			qdel(I)
+		qdel(D)
+
+	if(error_count)
+		fail("[error_count] design error(s) found. Every research design should cost more than what its product is worth when recycled.")
+	else
+		pass("All [tested_count] research designs with recyclable products have correct material costs.")
+
+	return 1
+
+
+datum/unit_test/stack_recipe_cost
+	name = "RECIPES: Stack Recipes"
+
+datum/unit_test/stack_recipe_cost/start_test()
+	var/tested_count = 0
+	var/error_count = 0
+	for(var/T in subtypesof(/material))
+		var/material/D = new T
+		var/list/recipe_lists = D.get_recipes()
```

Comment:
```suggestion
		var/list/datum/stack_recipe_list/recipe_lists = D.get_recipes()
```

Reply:
I totally forgot that's a thing. Neat, thanks!